.github/workflows/ruff-code-checks-and-poetry-cecks.yml

```diff
@@ -0,0 +1,24 @@
+name: Ruff code checks and poetry checks
+on: push
+
+jobs:
+  ruff-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - uses: chartboost/ruff-action@v1
```

Comment:
I don't like systems where it auto commits some changes. We already have pre-commit hook where it runs `ruff fix`.
This gh action should detect if code is not formatted/has errors and then it should raise error.